gnome-panel/gnome-panel-add-launcher.1
.\" This manpage has been automatically generated by docbook2man 
.\" from a DocBook document.  This tool can be found at:
.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
.\" Please send any bug reports, improvements, comments, patches, 
.\" etc. to Steve Cheng <steve@ggi-project.org>.
.TH "GNOME-PANEL-ADD-LAUNCHER" "1" "21 oktober 2001" "" ""
.SH NAME
gnome-panel-add-launcher \- add a launched in the Gnome panel.
.SH SYNOPSIS

\fBgnome-panel-add-launcher\fR [ \fB --panel \fI[=NUMBER]\fB\fR]  [ \fB --pos \fI[=NUMBER]\fB\fR]  [ \fB --url \fR] 

.SH "DESCRIPTION"
.PP
\fBgnome-panel-add-launcher\fR allows you to add a launcher in the
Gnome panel. Generally never used from the command line, except if you
know what you are doing.
.SH "OPTIONS"
.TP
\fB--panel NUMBER\fR
Panel to add the launcher to.
.TP
\fB--pos NUMBER\fR
Position to add the launcher to.
.TP
\fB--url\fR
The argument is a url to add, not a .desktop file.
.SH "AUTHOR"
.PP
This manual page was written by Christian Marillat <marillat@debian.org> for
the Debian GNU/Linux system (but may be used by others).


NEWS
===============
Version 2.17.90
===============

  Panel

	* Remove code warning (JP Rosevear)
	* Change text of some menu items (Luca Ferretti)
	* Use GTK_STOCK_MISSING_IMAGE when the icon of a launcher can't be
	  found and remove useless error dialog (Thomas Wood)
	* Use better filenames when creating launchers (Dan Winship, Vincent)
	* Simplify system menu code thanks to new control center shell
	  (Vincent)
	* Fix "Add to panel" dialog to correctly display capplets (Vincent)
	* Migrate settings menu buttons to a control center shell launcher
	  (Vincent)
	* Use 22 as default icon sizes for menus (Vincent)
	* Force size of launchers to 16, 22, 32 or 48 (Vincent)
	* Use new launcher icon (Vincent)
	* Fix compilation on darwin (Vincent)
	* Fix disappearing launcher on drag and drop (and other related issues)
	  (Vincent)
	* Fix transparency broken when screen size changes (bugreports@nn7.de)
	* Don't accept every command line option, but only the deprecated
	  --profile (Vincent)
	* Check for the logout_prompt gconf key to know whether we should show
	  the logout dialog or just log out (Ray Strode, Vincent)
	* Do not start dragging an applet when it's not possible because a grab
	  is active, fix change of cursor icon in this case (Amnon Aaronsohn)
	* Do not hardcode invisible_char in glade files (Vitaliy Ischenko)
	* Correctly position applet menu in RTL environments (Vincent)
	* Really remove launcher file when deleting a launcher, and do it only
	  when possible (Vincent)
	* Fix main menu not completely reloading when the menu is updated
	  (Vincent)

  gnome-desktop-item-edit

	* Make it possible to create a .directory file with a name instead of
	  forcing the name to $(dir)/.directory (Vincent)
	* Use new launcher icon (Vincent)
	* Make --create-new usage more logical (Vincent)

  libpanel-applet

	* Do not hardcode invisible_char in glade files (Vitaliy Ischenko)
	* Correctly position applet menu in RTL environments (Vincent)

  Clock Applet

	* Look for evolution binary before trying to run it (Vincent)
	* Make sure that the last day of a multi-day events is marked in the
	  calendar Michael Frank)
	* Don't mark multiple days on the calendar if the event is an all day
	  event (Vincent)
	* Fix compilation on some systems (Vincent)
	* Launch evolution when double-clicking on a task (Nickolay V. Shmyrev,
	  Vincent)

  Fish Applet

	* HIG fix (Vincent)
	* Do not hardcode invisible_char in glade files (Vitaliy Ischenko)

  Notification Area Applet

	* Add a test-tray program (Christian Persch)

  Workspace Switcher Applet

	* Slow down speed of the spin button for number of workspaces
	  (Manu Cornet)

  Misc

	* New launcher icon
	* Build fixes (Vincent)

  Documentation

	* Add a link to the user guide (Claude Paroz)

  Docs Translators

	* Nickolay V. Shmyrev (ru)

  Translators

	* Khaled Hosny (ar)
	* Ihar Hrachyshka (be)
	* Jordi Mallach (ca)
	* Jakub Friedl (cs)
	* Hendrik Richter (de)
	* David Lodge (en_GB)
	* Ivar Smolin (et)
	* Yuval Tanny (he)
	* Gabor Kelemen (hu)
	* Luca Ferretti (it)
	* A S Alam (pa)
	* Данило Шеган (sr)
	* Daniel Nylander (sv)
	* Theppitak Karoonboonyanan (th)

==============
Version 2.16.2
==============

  Panel

	* Fix error when removing launcher (Tom Tromey)
	* HIG fixes (Dennis Cranston)
	* Fix leaks (Kjartan Maraas, Paolo Borelli, Vincent)
	* Fix a warning string (Claudio Saavedra)

  Notification Area Applet

	* Fix use of uninitialized memory (JP Rosevear)

  Misc

	* Check for libXau (Matt Keenan, Vincent)
	* Fix doc build (Kjartan Maraas)

  Docs Translators

	* Francisco Javier F. Serrador (es)

  Translators

	* Djihed Afifi (ar)
	* Jakub Friedl (cs)
	* Christian Kintner (de)
	* Nikos Charonitakis (el)
	* Francisco Javier F. Serrador (es)
	* Ivar Smolin (et)
	* Ilkka Tuohela (fi)
	* Ignacio Casal Quinteiro (gl)
	* Satoru SATOH (ja)
	* Åsmund Skjæveland (nn)

==============
Version 2.16.1
==============

  Panel

	* Fix crash when alacarte isn't installed (Frédéric Crozat)
	* Do not present the 'new launcher' in the "add to panel" dialog
	  when the command line is disabled (Bastien Nocera)

  Misc

	* Require libecal 1.6.0 (Vincent)

  Docs Translators

	* Luca Ferretti (it)
	* Daniel Nylander (sv)

  Translators

	* Vladimir Petkov (bg)
	* Jakub Friedl (cs)
	* Christian Kintner (de)
	* Mindu Dorji (dz)
	* David Lodge (en_GB)
	* Ivar Smolin (et)
	* Ilkka Tuohela (fi)
	* Luca Ferretti (it)
	* Wouter Bolsterlee (nl)
	* Subhransu Behera (or)

==============
Version 2.16.0
==============

  Panel

	* Plug leaks (Paolo Borelli)
	* Don't (wrongly) modify the laucnher when creating the editor
	  (Vincent)

  Clock Applet

	* Fix crash when double-clicking on the calendar (Vincent)

  Misc

	* Add 24x24 versions of the icons (Vincent)

  Translators

	* Vladimir Petkov (bg)
	* Runa Bhattacharjee (bn_IN)
	* Jordi Mallach (ca)
	* Nikos Charonitakis (el)
	* Francisco Javier F. Serrador (es)
	* Iñaki Larrañaga Murgoitio (eu)
	* Robert-André Mauchin (fr)
	* Rajesh Ranjan (hi)
	* Gabor Kelemen (hu)
	* Raivis Dejus (lv)
	* Jovan Naumovski (mk)
	* Ani Peter (ml)
	* Rahul Bhalerao (mr)
	* Subhransu Behera (or)
	* Evandro Fernandes Giovanini (pt_BR)
	* Duarte Loreto (pt)
	* Горан Ракић (sr)
	* Priya (ta)
	* Theppitak Karoonboonyanan (th)
	* Woodman Tuen (zh_HK)
	* Woodman Tuen (zh_TW)

===============
Version 2.15.92
===============

  Panel

	* Fix panel going to the top of screen instead of autohiding
	  (Зоран Рилак)
	* Make relative paths work in gnome-desktop-item-edit (Vincent)
	* Use new icon for run dialog (Vincent)
	* Fix crash when alacarte is not found for editing a menu (Ray Strode)
	* Handle drop of some special icons on the nautilus desktop (trash,
	  home, computer, network) (Vincent)
	* Fix drag and drop of items from the menu (Vincent)
	* Remove extensions of the icon name before looking for it for the
	  menus (Vincent)
	* Don't add ellipsises at the end of "Delete Panel" (Vincent)
	* Update icon for the network places (Vincent)

gnome-panel/gnome-panel-add-launcher.c
/*   gnome-panel-add-launcher: Thingie to add a launcher
 *
 *   Copyright 2000 Eazel, Inc.
 *   Authors: George Lebl <jirka@5z.com>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
 * USA
 */

#include <config.h>
#include <libgnomeui/libgnomeui.h>
#include <bonobo-activation/bonobo-activation.h>
#include <libgnomeui/gnome-window-icon.h>
#include <gdk/gdkx.h>
#include "gnome-panel.h"

static int panel = 0;
static int pos = 0;
static gboolean url = FALSE;

static const struct poptOption options[] = {
	{ "panel", 0, POPT_ARG_INT, &panel, 0, N_("Panel to add the launcher to"), N_("NUMBER") },
	{ "pos", 0, POPT_ARG_INT, &pos, 0, N_("Position to add the launcher to"), N_("NUMBER") },
	{ "url", 0, POPT_ARG_NONE, &url, 0, N_("The argument is a url to add, not a .desktop file"), NULL },
	{ NULL } 
};

int
main (int argc, char **argv)
{
	CORBA_Environment ev;
	poptContext ctx;
	const char **args;
	const char *arg;
	GNOME_Panel2 panel_client = CORBA_OBJECT_NIL;

	bindtextdomain (PACKAGE, GNOMELOCALEDIR);
	textdomain (PACKAGE);

	CORBA_exception_init(&ev);

	if (gnome_CORBA_init_with_popt_table ("gnome-panel-add-launcher", VERSION,
					      &argc, argv,
					      options, 0, &ctx,
					      GNORBA_INIT_SERVER_FUNC, &ev) == NULL)
		return 1;

	args = poptGetArgs (ctx);

	if (args == NULL ||
	    args[0] == NULL ||
	    args[1] != NULL) {
		fprintf (stderr,
			 _("You must supply a single argument with the "
			   ".desktop file or url to use\n"));
		return 1;
	}

	arg = args[0];

	panel_client =
		goad_server_activate_with_repo_id (NULL,
						   "IDL:GNOME/Panel2:1.0",
						   GOAD_ACTIVATE_EXISTING_ONLY,
						   NULL);
	
	if (panel_client == NULL) {
		fprintf (stderr, _("No panel found\n"));
		return 1;
	}

	if (url)
		GNOME_Panel2_add_launcher_from_info_url (panel_client, arg, arg, arg, "", panel, pos, &ev);
	else
		GNOME_Panel2_add_launcher (panel_client, arg, panel, pos, &ev);

	CORBA_exception_free(&ev);

	return 0;
}
